lib/index.d.ts

```diff
@@ -0,0 +1,528 @@
+/// <reference types="@panva/jose" />
+/// <reference types="node" />
+
+/**
+ * @see https://github.com/panva/node-openid-client/blob/master/docs/README.md
+ */
+declare module 'openid-client' {
+  import { JWKS } from '@panva/jose'
+  import { IncomingMessage } from 'http'
+  import { Http2ServerRequest } from 'http2'
+
+  /**
+   * @see https://medium.com/@darutk/diagrams-of-all-the-openid-connect-flows-6968e3990660
+   */
+  enum AuthResponseType {
+    CODE = 'code',
+    TOKEN = 'token',
+    ID_TOKEN = 'id_token',
+  }
+
+  /**
+   * @see https://github.com/panva/node-openid-client/blob/master/docs/README.md#client-authentication-methods
+   */
+  enum TokenAuthMethod {
```

Comment:
If we're defining ambient enums in the declaration file, they'll also need to exist in the javascript to be useable